Diplomatic/Utils/Converters/RegexMatch.cs

```diff
@@ -0,0 +1,24 @@
+﻿using System;
+using System.Globalization;
+using System.Text.RegularExpressions;
+using Xamarin.Forms;
+
+namespace Diplomatic.Utils.Converters
+{
+    public class RegexMatch : IValueConverter
+    {
+        public string Pattern { get; set; } = "";
+
+        public object Convert(object value, Type targetType, object parameter, CultureInfo culture)
+        {
+            string text = (string)value;
+            Regex regex = new Regex(Pattern);
+            return regex.IsMatch(text);
```

Comment:
You should replace this inefficient code with a call to the static [Regex.IsMatch(String, String)](https://docs.microsoft.com/en-us/dotnet/api/system.text.regularexpressions.regex.ismatch#System_Text_RegularExpressions_Regex_IsMatch_System_String_System_String_) method.